durhamatletico/durhamatletico-cms
# Theme updates

It's happening! 👐  Here's a giant PR for your review at your convenience!

## Summary

I overhauled pretty much the entire theme! Some details:

- Updated colors, fonts, typography styles
- New header
- New featured item on homepage
- Tons of style updates to the game views
- New /posts layout and updates to individual post pages
- Extended /post styles to taxonomy term pages
- Updated sponsor and footer styles

## Information you need

### To test

- Pull this branch and run `cex`, `updb`, and `cr`
- Follow the theme README instructions for installing the node project and compiling CSS. Compile CSS locally.
- One manual deployment step: I couldn't get the featured item default image to export to configuration. You'll need to go [here](https://local.durhamatletico.com/admin/structure/types/manage/featured_item/fields/node.featured_item.field_featured_item_image) and set `web/themes/custom/atletico/images/default-banner-image.jpg` as the default image. The home page layout will be wonky until you do this.
- Please review the site thoroughly, especially any pages you think I may have missed
- Feedback very welcome!! I can spend some more time before my trip on the 29th making updates, otherwise I can work on it once I'm back. Let me know if you find any bugs or if there are any changes you'd like to make. I want to make sure you and David are happy with things.

Note that I'm still planning on doing some cross-browser and accessibility testing.

### Featured Item

- I added an image field to the featured item content type. Now you easily change the banner image on the front page.
- Once you set the default image as noted above, you can either use that or include a different image for each node
- You can include two CTA links if you want to, although I wouldn't really recommend it. I'm not printing the "description" field to the page anywhere but we could add this somewhere if you want.

### Posts

- I'd encourage you guys to use the image field if possible; this improves the look of the latest post view on the home page and the views on /posts. Images within the body text may or may not show in these views since only the first 600 characters of the body are shown, plus images take up a lot of characters so if there's an image there probably won't be much text, which looks odd. The actual image field guarantees that there will be image and a decent amount of text showing in the teaser view.
- I'd also encourage you to use headings in order - `<h2>` elements have a special style in articles (similar to the page title underline style) which breaks up the text nicely
- Previously, all images were centered via CSS in posts (except images that were explicitly floated to the left or right). I removed that style so you'll need to position images within the body of posts. This shouldn't have much of an impact on existing posts, just letting you know for the future.

### Submenus

Unfortunately I couldn't fix the issue with submenus only showing on their parent link's page. The "Show as expanded" checkbox isn't being exported to configuration for some reason. You could update the 3 menu items with submenus [here](https://local.durhamatletico.com/admin/structure/menu/manage/main) as a manual deployment step. This wouldn't be a huge pain if you're not deploying very often.

## Questions

- I removed the Gemfile and compass config without thinking that this might affect your ability to compile CSS for production - should I add those back?
- Spring 2016 Registration is showing on user profile for non-admins - should that block be removed?
- Did you read this far? Kudos 👏 

## Screenshot

![screencapture-local-durhamatletico-2018-08-21-20_02_50](https://user-images.githubusercontent.com/8465998/44435275-36948c00-a57d-11e8-9e22-5c183f7a8ded.png)
